### ISSUE TITLE
View Password icon (eyeball) isn't keyboard navigable

### ISSUE THREAD

**Issue by Filupnot** [medplum/medplum#9786]
In the registration workflow, you're unable to use a keyboard to navigate to the eyeball icon that allows you to view your password. It's not part of the tab order and can't be accessed via arrow keys (or other keyboard methods). 

Claude thinks that this can be fixed with a Mantine component toggle. 

https://github.com/user-attachments/assets/54243dcf-86da-4099-b6f2-3b01339ead0e

**Comment by Filupnot** [comment]
Looks like this was actually a deliberate decision by Mantine in 2021: https://github.com/orgs/mantinedev/discussions/131

That being said, this violates [WCAG 2.1.1,](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html) so I think this still worth changing here, especially given our healthcare user base. Also I don't agree with this; keyboard navigation isn't just used by screenreader users:
> if you're using a screen-reader then I am sure it's handling that field in a special way anyway. (so the visibility toggle is maybe not helpful?)
